registry/storage/registry.go

```diff
@@ -100,7 +104,12 @@ func (repo *repository) Name() string {
 // may be context sensitive in the future. The instance should be used similar
 // to a request local.
 func (repo *repository) Manifests() distribution.ManifestService {
-	return &manifestStore{
+	tomb := tomb{
```

Comment:
This can be instantiated in the `Repository` method and shared with `Manifests` and `Blobs`.